File: app/src/main/java/trunk/doi/base/bean/GankItemData.java
package trunk.doi.base.bean;

import android.os.Parcel;
import android.os.Parcelable;



/**
 * Author: Othershe
 * Time: 2016/8/12 14:34
 */


public class GankItemData implements Parcelable {



    private String _id;

    private String createdAt;
    private String desc;
    private String[] images;
    private String publishedAt;
    private String source;
    private String type;
    private String url;

    private boolean used;
    private String who;





    public String get_id() {
        return _id;
    }

    public void set_id(String _id) {
        this._id = _id;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    public String getDesc() {
        return desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    public String[] getImages() {
        return images;
    }

    public void setImages(String[] images) {
        this.images = images;
    }

    public String getPublishedAt() {
        return publishedAt;
    }

    public void setPublishedAt(String publishedAt) {
        this.publishedAt = publishedAt;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public boolean isUsed() {
        return used;
    }

    public void setUsed(boolean used) {
        this.used = used;
    }

    public String getWho() {
        return who;
    }

    public void setWho(String who) {
        this.who = who;
    }



    public GankItemData() {
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this._id);
        dest.writeString(this.createdAt);
        dest.writeString(this.desc);
        dest.writeStringArray(this.images);
        dest.writeString(this.publishedAt);
        dest.writeString(this.source);
        dest.writeString(this.type);
        dest.writeString(this.url);
        dest.writeByte(this.used ? (byte) 1 : (byte) 0);
        dest.writeString(this.who);
    }

    protected GankItemData(Parcel in) {
        this._id = in.readString();
        this.createdAt = in.readString();
        this.desc = in.readString();
        this.images = in.createStringArray();
        this.publishedAt = in.readString();
        this.source = in.readString();
        this.type = in.readString();
        this.url = in.readString();
        this.used = in.readByte() != 0;
        this.who = in.readString();
    }

    public static final Parcelable.Creator<GankItemData> CREATOR = new Parcelable.Creator<GankItemData>() {
        @Override
        public GankItemData createFromParcel(Parcel source) {
            return new GankItemData(source);
        }

        @Override
        public GankItemData[] newArray(int size) {
            return new GankItemData[size];
        }
    };
}

File: app/src/main/java/com/gas/app/bean/GankItemData.kt
package com.gas.app.bean

import android.os.Parcel
import android.os.Parcelable

/**
 *
 */
class GankItemData : Parcelable {
    private var _id: String? = null
    var createdAt: String? = null
    var desc: String? = null
    var images: Array<String>? = null
    var source: String? = null
    var type: String? = null
    var publishedAt: String? = null
    var url: String? = null
    var isUsed = false
    var who: String? = null
    fun get_id(): String? {
        return _id
    }

    fun set_id(_id: String?) {
        this._id = _id
    }

    constructor() {}

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(_id)
        dest.writeString(createdAt)
        dest.writeString(desc)
        dest.writeStringArray(images)
        dest.writeString(publishedAt)
        dest.writeString(source)
        dest.writeString(type)
        dest.writeString(url)
        dest.writeByte(if (isUsed) 1.toByte() else 0.toByte())
        dest.writeString(who)
    }

    protected constructor(`in`: Parcel) {
        _id = `in`.readString()
        createdAt = `in`.readString()
        desc = `in`.readString()
        images = `in`.createStringArray()
        publishedAt = `in`.readString()
        source = `in`.readString()
        type = `in`.readString()
        url = `in`.readString()
        isUsed = `in`.readByte().toInt() != 0
        who = `in`.readString()
    }

    override fun toString(): String {
        return "GankItemData{" +
                "_id='" + _id + '\'' +
                ", desc='" + desc + '\'' +
                ", publishedAt='" + publishedAt + '\'' +
                ", url='" + url + '\'' +
                ", who='" + who + '\'' +
                '}'
    }

    companion object {
        val CREATOR: Parcelable.Creator<GankItemData> = object : Parcelable.Creator<GankItemData> {
            override fun createFromParcel(source: Parcel): GankItemData? {
                return GankItemData(source)
            }

            override fun newArray(size: Int): Array<GankItemData?> {
                return arrayOfNulls(size)
            }
        }
    }
}